src/tetris/profiler/service/fppAnalysisService.ts

```diff
@@ -58,7 +56,7 @@ export default class FppAnalysisService extends BaseService {
 	//region protected methods
 	protected async _run(successCallback: (senderName: string, measurement: Measurement<Object>) => void): Promise<void> {
 		this._successCallback = successCallback;
-		if (CameraController.instance.permissionState !== HardwarePermission.granted) {
+		if (!await CameraController.instance.requestWebcamPermissions()) {
```

Comment:
This is no substitute for checking wether or not the permission has been granted. Is this intended?

Reply:
`requestWebcamPermission` actually returns a boolean, indicating whether or not the priviliges are given or not.
This also includes previously granted permissions and any kind of failure (as in "permission not granted (for whatever reason"), which is just what we want at that point.